aggregate_root/lib/aggregate_root.rb

```diff
@@ -4,6 +4,22 @@
 require 'aggregate_root/default_apply_strategy'
 
 module AggregateRoot
+  module ClassMethods
+    def on(*event_klasses, apply_strategy:  DefaultApplyStrategy.new, &block)
+      event_klasses.each do |event_klass|
+        handler_name = apply_strategy.handler_name_by_class(event_klass)
```

Comment:
this seems _weird_ to me... Could we not `tell` the strategy that we defined a method (with any name, any convention) and we want it to be used as a handler for `*event_klasses` instead of asking the `apply_strategy` about this the method name we should define?